src/datasources.js

```diff
@@ -128,11 +134,13 @@ define([
                     success: finish,
                     error: function (jqXHR, errorType, exc) {
                         finish([]);
-                        if (errorCallback) {
+                        _.each(errorCallbacks, function(callback) {
                             errorCallback(jqXHR, errorType, exc);
```

Comment:
I think this should be `callback(jqXHR, errorType, exc);`

Also, might want to check if `callback` is not undefined before calling it.